src/careamics/careamist.py

```diff
@@ -643,16 +634,41 @@ def predict(
                 # record datamodule
                 self.pred_datamodule = datamodule
 
-                return self.trainer.predict(
-                    model=self.model, datamodule=datamodule, ckpt_path=checkpoint
-                )
-
             else:
                 raise ValueError(
                     f"Invalid input. Expected a CAREamicsWood instance, paths or "
                     f"np.ndarray (got {type(source)})."
                 )
 
+        # TODO: tidy up: move to a seperate function
+        predictions = self.trainer.predict(
+            model=self.model, datamodule=self.pred_datamodule, ckpt_path=checkpoint
+        )
+        # not sure if necessary
+        if len(predictions) == 0:
+            return predictions
+        is_tiled = len(predictions[0]) == 2
+        if is_tiled:
+            # "de-collate"
+            tile_infos = [
+                tile_info
+                for _, tile_info_list in predictions
+                for tile_info in tile_info_list
+            ]
+            predictions = torch.concatenate([preds for preds, _ in predictions])
+            # convert to numpy
+            predictions = predictions.cpu().numpy()
+            predictions = stitch_prediction(predictions, tile_infos)
+        else:
+            # convert to numpy
+            predictions = [pred.cpu().numpy() for pred in predictions]
```

Comment:
So not sure whether this happens somewhere else in the code, but if the predictions have all the same size we could concatenate them in a single array.

Reply:
I am changing a lot of stuff, this PR is actually a proper draft. I have put it up more to get an opinion on applying tiling after prediction before I put a lot of work in! So this is pretty rough and minimally functioning